runtime/js/11_workers.js

```diff
@@ -139,7 +139,7 @@
 
     #pollControl = async () => {
       while (this.#status === "RUNNING") {
-        const [type, data] = await hostRecvCtrl(this.#id);
+        const { 0: type, 1: data } = await hostRecvCtrl(this.#id);
```

Comment:
I don't really get these changes, @petamoriken could you explain what's the advantage of using this syntax over array destructuring?

Reply:
Okay, I'm asking because I see changes in test files that are a bit surprising - these tests should ideally be checked with exactly same rules as user code since they are testing public API that users would use. Do you think we should revert changes to test code?

Reply:
IMO, modern `for-of` should be used in test codes, not `for-in`, which has traps. I believe it is worthwhile to apply the guard-for-in rule to test codes.